source/taurus/enumify/package.d

```diff
@@ -0,0 +1,255 @@
+module taurus.enumify;
+
+import localimport;
+
+/**
+Defines a new member. `Args` follows the construction format of `Tuple`.
+
+Params:
+	name = member name
+	Args = types and names(optional)
+
+Examples:
+---
+enum Foo { A, B }
+@Member!("A")
+@Member!("B", int, "name")
+@Member!("C", byte, long)
+@Member!("D", float, char, "name")
+@Member!("E", Foo, "nameA", Tuple!(uint, uint), "nameB")
+---
+*/
+struct Member(Args ...)
+	if (Args.length > 0
+		&& from.std.traits.isSomeString!(typeof(Args[0]))
+		&& __traits(compiles, { static if (Args.length > 1) from.std.typecons.Tuple!(Args[1 .. $]) t; })
+	)
+{}
+
+@safe pure nothrow @nogc unittest {
+	assert(!__traits(compiles, Member!()));
+	assert(!__traits(compiles, Member!(int)));
+	assert(!__traits(compiles, Member!("name", "foo")));
+	assert(!__traits(compiles, Member!("name", int, "foo", "bar")));
+	assert(!__traits(compiles, Member!("name", int, "foo", short, "foo")));
+
+	assert( __traits(compiles, Member!("A")));
+	assert( __traits(compiles, Member!("B", int, "name")));
+	assert( __traits(compiles, Member!("C", byte, long)));
+	assert( __traits(compiles, Member!("D", float, char, "name")));
+	assert( __traits(compiles, Member!("E", int, "nameA", from.std.typecons.Tuple!(uint, uint), "nameB")));
+}
+
+/**
+Generate an enum like struct of dynamic values.
+
+Examples:
+---
+// TODO: add Message example
+---
+
+GenAttributes:
+// TODO: add generic generated attr
+*/
+mixin template enumify(from.std.typecons.Flag!"defautToString" defautToString = from.std.typecons.Flag!"defautToString".yes)
+{
+	// takes all attributes used before the mixin call
+	private struct __udahelper() {}
```

Comment:
Please use only one `_` as two are reserved by the language specification.

> 2.8.2. Implementation Defined: Identifiers starting with __ (two underscores) are reserved.

Reply:
done